# **Customer Churn Prediction** 
'Churn' - It is the measure of how many customers stop using a product.

## **Introduction**
In the competitive landscape of business, maximizing profit margins and ensuring customer retention are paramount objectives for companies across industries. Leveraging the power of machine learning, organizations are increasingly turning to classification models to conduct churn analysis on their customer base. Classification, a fundamental concept in machine learning and statistics, involves a supervised learning approach where algorithms learn from existing data to predict and classify new observations.  

## **Business Understanding Statement**
In today's competitive business environment, maximizing profit margins and retaining customers are top priorities for companies across industries. Leveraging the capabilities of machine learning, organizations are increasingly adopting classification models for churn analysis, a critical aspect of customer relationship management. Classification models, rooted in machine learning and statistics, enable businesses to predict and categorize customers based on their likelihood of churning, thereby empowering proactive retention strategies and sustainable growth.
## **Objective**
The objective of this project is to develop and deploy robust classification models for customer churn analysis using advanced machine learning techniques. Specifically, my goal is to:
1. Utilize historical customer data to train and validate classification algorithms that accurately predict customer churn probabilities.
2. Implement scalable and interpretable machine learning models that can categorize customers into churn-risk segments for targeted retention efforts.
3. Optimize model performance by fine-tuning hyperparameters, feature selection, and model evaluation metrics to achieve high accuracy and predictive power.
4. Integrate the developed models into existing business processes, enabling real-time churn prediction and actionable insights for decision-makers.
5. Evaluate the effectiveness of the deployed models through continuous monitoring, validation, and refinement to ensure long-term sustainability and impact on customer retention and profitability.  

By achieving these objectives, I aim to empower businesses with actionable insights derived from data-driven churn analysis, enabling them to implement proactive retention strategies, optimize resource allocation, and drive sustainable growth in today's competitive market landscape.
## **Project Goals(KPIs)**
- Develop accurate classification models for churn prediction.
- Optimize retention strategies based on model insights.
- Drive sustainable growth and profitability by reducing churn rates.
- Provide actionable insights for informed decision-making.
- Enhance customer experience and engagement.

### Importing Necessary Packages

In [26]:
import pyodbc    
from dotenv import dotenv_values
import pandas as pd
import warnings 

warnings.filterwarnings('ignore')

### Loading Environment variables

In [27]:
# Load environment variables from .env file into a dictionary
environment_variables = dotenv_values('.env')
# Get the values for the credentials you set in the '.env' file
server = environment_variables.get("SERVER")
database = environment_variables.get("DATABASE")
username = environment_variables.get("USERNAME")
password = environment_variables.get("PASSWORD")

connection_string = f"DRIVER={{SQL Server}};SERVER={server};DATABASE={database};UID={username};PWD={password};MARS_Connection=yes;MinProtocolVersion=TLSv1.2;"

In [28]:
connection = pyodbc.connect(connection_string)

#### Geting the Training data
Using SQL to generate the table

In [29]:
# Selecting data using the SQL syntax
query = "Select * from dbo.LP2_Telco_churn_first_3000"
# Reading the first part of training data into a data frame
data1 = pd.read_sql(query, connection)

#### Getting the second Training dataset
The second dataset is downloaded to my local drive and loaded to the python environment

In [30]:
# Loading second training data
data2 = pd.read_csv("C:/Users/EWamathai/Desktop/Customer Churn Prediction/Customer-Churn-Prediction/files/LP2_Telco-churn-second-2000.csv")

### Concating the training datasets

In [31]:
# Loading the two dataframes into a variable
data3 = [data1, data2]
# Concatenation of the data frames
train_df = pd.concat(data3, ignore_index=True)

In [32]:
# Checking training data info
train_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5043 entries, 0 to 5042
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customerID        5043 non-null   object 
 1   gender            5043 non-null   object 
 2   SeniorCitizen     5043 non-null   int64  
 3   Partner           5043 non-null   object 
 4   Dependents        5043 non-null   object 
 5   tenure            5043 non-null   int64  
 6   PhoneService      5043 non-null   object 
 7   MultipleLines     4774 non-null   object 
 8   InternetService   5043 non-null   object 
 9   OnlineSecurity    4392 non-null   object 
 10  OnlineBackup      4392 non-null   object 
 11  DeviceProtection  4392 non-null   object 
 12  TechSupport       4392 non-null   object 
 13  StreamingTV       4392 non-null   object 
 14  StreamingMovies   4392 non-null   object 
 15  Contract          5043 non-null   object 
 16  PaperlessBilling  5043 non-null   object 


## **Data-Frame Understanding**  
**Data Integrity**  
The DataFrame contains 5043 entries across 21 columns and thus a substantial dataset for analysis. Most columns have non-null counts indicating complete data, but there are missing values in 
- MultipleLines
- OnlineSecurity 
- OnlineBackup
- DeviceProtection
- TechSupport
- StreamingTV
- StreamingMovies
- TotalCharges
- Churn  

**Data Types**  
- The 'SeniorCitizen' column is represented as an integer, indicating a binary variable (likely indicating whether the customer is a senior citizen or not).  
- 'MonthlyCharges' is correctly represented as a float, but 'TotalCharges' is shown as an object, which may need conversion to numeric for numerical analysis.  

**Categorical Variables**  
- Several columns such as 'gender', 'Partner', 'Dependents', 'PhoneService', 'MultipleLines', 'InternetService', 'OnlineSecurity', 'OnlineBackup', 'DeviceProtection', 'TechSupport', 'StreamingTV', 'StreamingMovies', 'Contract', 'PaperlessBilling', 'PaymentMethod', and 'Churn' are categorical variables, likely representing customer characteristics, services subscribed to, and contract details.  
- The 'Churn' column is the target variable indicating whether a customer churned or not.  

**Numeric Variables**  
- 'tenure' represents the number of months a customer has stayed with the company, which can be a significant predictor of churn.  
- 'MonthlyCharges' and 'TotalCharges' are likely related to billing information and customer spending, which can also be important predictors of churn.  

**Data Cleaning and Preprocessing**  
- Missing values in 'MultipleLines', 'OnlineSecurity', 'OnlineBackup', 'DeviceProtection', 'TechSupport', 'StreamingTV', 'StreamingMovies', 'TotalCharges', and 'Churn' may require imputation or removal depending on their impact on the analysis.  
- The 'TotalCharges' column should be converted to a numeric data type for numerical computations.  

**Churn Analysis**  
Since the goal is to predict customer churn, analyzing the distribution of 'Churn' values (e.g., churned vs. non-churned customers) and exploring relationships between predictor variables and churn status will be crucial.  

**Feature Engineering**  
Creating new features or deriving insights from existing ones (e.g., customer tenure in months, total spending) can enhance the predictive power of the model.

## **Business Questions**
- **Demographic Analysis**
1. How does customer churn vary based on demographic factors such as gender, age (SeniorCitizen), marital status (Partner), and presence of dependents (Dependents)?
2.  Are there any specific demographic segments that are more prone to churn than others?
- **Tenure Analysis**
1. What is the relationship between customer tenure (length of subscription) and churn rate?
2.  Do long-tenured customers exhibit lower churn rates compared to new customers?
- **Service Usage Analysis**
1. How does the availability and utilization of services such as PhoneService, MultipleLines, InternetService, OnlineSecurity, OnlineBackup, DeviceProtection, TechSupport, StreamingTV, and StreamingMovies impact customer churn?
2. Are customers subscribing to additional services less likely to churn?
- **Contract Analysis**
1. What is the churn rate among customers based on different contract types (Contract)?
2. Do customers with longer-term contracts exhibit lower churn rates compared to those with month-to-month contracts?
- **Billing Preferences**
1. Does the preference for paperless billing (PaperlessBilling) influence customer churn?
2. Are customers using certain payment methods (PaymentMethod) more likely to churn than others?
- **Financial Impact**
1. How do monthly charges (MonthlyCharges) and total charges (TotalCharges) affect customer churn?
2. Do customers with higher monthly charges or total charges have different churn behavior?
- **Customer Support Analysis**
1. What is the impact of customer support services (OnlineSecurity, OnlineBackup, DeviceProtection, TechSupport) on reducing churn?
2. Are customers who utilize tech support or security services less likely to churn?
- **Correlation Analysis**
1. Are there any significant correlations between different features (e.g., tenure, monthly charges, contract type) and customer churn?
2. Which factors have the strongest correlation with churn behavior?

## **Hypothesis Testing** 
**Null Hypothesis (H0):** The tenure of customers does not have a significant impact on the likelihood of customer churn in the company.  
**Alternative Hypothesis (H1):** The tenure of customers significantly affects the likelihood of customer churn in the company.   

This hypothesis focuses specifically on the impact of customer tenure on churn behavior

## **Data Cleaning and Preprocessing**

In [33]:
# Checking the training data frame
train_df.head()

customerID  gender  SeniorCitizen Partner Dependents  tenure PhoneService  \
0  7590-VHVEG  Female              0    True      False       1        False   
1  5575-GNVDE    Male              0   False      False      34         True   
2  3668-QPYBK    Male              0   False      False       2         True   
3  7795-CFOCW    Male              0   False      False      45        False   
4  9237-HQITU  Female              0   False      False       2         True   

  MultipleLines InternetService OnlineSecurity  ... DeviceProtection  \
0          None             DSL          False  ...            False   
1         False             DSL           True  ...             True   
2         False             DSL           True  ...            False   
3          None             DSL           True  ...             True   
4         False     Fiber optic          False  ...            False   

  TechSupport StreamingTV StreamingMovies        Contract PaperlessBilling  \
0       False       False           False  Month-to-month             True   
1       False       False           False        One year            False   
2       False       False           False  Month-to-month             True   
3        True       False           False        One year            False   
4       False       False           False  Month-to-month             True   

               PaymentMethod MonthlyCharges  TotalCharges  Churn  
0           Electronic check      29.850000         29.85  False  
1               Mailed check      56.950001        1889.5  False  
2               Mailed check      53.849998    108.150002   True  
3  Bank transfer (automatic)      42.299999       1840.75  False  
4           Electronic check      70.699997    151.649994   True  

[5 rows x 21 columns]

#### **Cleaning 'TotalCharges' Column**

In [34]:
# Checking total Charges column
train_df['TotalCharges'].info()

<class 'pandas.core.series.Series'>
RangeIndex: 5043 entries, 0 to 5042
Series name: TotalCharges
Non-Null Count  Dtype 
--------------  ----- 
5038 non-null   object
dtypes: object(1)
memory usage: 39.5+ KB


In [35]:
# Converting the data types for the total charges column
train_df['TotalCharges'] = pd.to_numeric(train_df['TotalCharges'], errors='coerce')

In [36]:
train_df.isna().sum().sort_values(ascending=False)

OnlineBackup        651
OnlineSecurity      651
StreamingMovies     651
StreamingTV         651
TechSupport         651
DeviceProtection    651
MultipleLines       269
TotalCharges          8
Churn                 1
InternetService       0
gender                0
PhoneService          0
tenure                0
Dependents            0
Partner               0
Contract              0
PaperlessBilling      0
PaymentMethod         0
MonthlyCharges        0
SeniorCitizen         0
customerID            0
dtype: int64

In [37]:
# Checking the nulls in the total charges column
train_df[train_df['TotalCharges'].isna()]

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
488   4472-LVYGI  Female              0    True       True       0   
753   3115-CZMZD    Male              0   False       True       0   
936   5709-LVOEQ  Female              0    True       True       0   
1082  4367-NUYAO    Male              0    True       True       0   
1340  1371-DWPAZ  Female              0    True       True       0   
3218  2923-ARZLG    Male              0     Yes        Yes       0   
4670  4075-WKNIU  Female              0     Yes        Yes       0   
4754  2775-SEFEE    Male              0      No        Yes       0   

     PhoneService MultipleLines InternetService       OnlineSecurity  ...  \
488         False          None             DSL                 True  ...   
753          True         False              No                 None  ...   
936          True         False             DSL                 True  ...   
1082         True          True              No                 None  ...   
1340        False          None             DSL                 True  ...   
3218          Yes            No              No  No internet service  ...   
4670          Yes           Yes             DSL                   No  ...   
4754          Yes           Yes             DSL                  Yes  ...   

         DeviceProtection          TechSupport          StreamingTV  \
488                  True                 True                 True   
753                  None                 None                 None   
936                  True                False                 True   
1082                 None                 None                 None   
1340                 True                 True                 True   
3218  No internet service  No internet service  No internet service   
4670                  Yes                  Yes                  Yes   
4754                   No                  Yes                   No   

          StreamingMovies  Contract PaperlessBilling  \
488                 False  Two year             True   
753                  None  Two year            False   
936                  True  Two year            False   
1082                 None  Two year            False   
1340                False  Two year            False   
3218  No internet service  One year              Yes   
4670                   No  Two year               No   
4754                   No  Two year              Yes   

                  PaymentMethod MonthlyCharges  TotalCharges  Churn  
488   Bank transfer (automatic)      52.549999           NaN  False  
753                Mailed check      20.250000           NaN  False  
936                Mailed check      80.849998           NaN  False  
1082               Mailed check      25.750000           NaN  False  
1340    Credit card (automatic)      56.049999           NaN  False  
3218               Mailed check      19.700000           NaN     No  
4670               Mailed check      73.350000           NaN     No  
4754  Bank transfer (automatic)      61.900000           NaN     No  

[8 rows x 21 columns]

<span style="color: orange;">**Observation**</span>  All the 8 nulls in the 'TotalCharges' column are with respect to rows whose data in the 'tenure' column are zero. and thus, I will fill the nulls in the TotalCharges column using the corresponding values in the monthly charges. This is under the presumption that the tenure being zero means the customer has either just joined and thus not yet into a full tenure.

In [38]:
# Fill null values in 'TotalCharges' with 'MonthlyCharges' where 'tenure' is 0
train_df['TotalCharges'] = train_df.apply(lambda row: row['MonthlyCharges'] 
        if pd.isnull(row['TotalCharges']) and row['tenure'] == 0 
        else row['TotalCharges'], axis=1)

In [39]:
print(f'The null values in the Total Charges column is now: {train_df['TotalCharges'].isnull().sum()}')

The null values in the Total Charges column is now: 0


#### **Filling the nulls in the Multiplelines Column**

In [40]:
# Checking Total Nulls in MultipleLines
print(f'The null values in the MultipleLines column is now: {train_df['MultipleLines'].isnull().sum()}')

The null values in the MultipleLines column is now: 269


In [41]:
train_df[train_df['MultipleLines'].isna()]

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
0     7590-VHVEG  Female              0    True      False       1   
3     7795-CFOCW    Male              0   False      False      45   
7     6713-OKOMC  Female              0   False      False      10   
20    8779-QRDMV    Male              1   False      False       1   
27    8665-UTDHZ    Male              0    True       True       1   
...          ...     ...            ...     ...        ...     ...   
2972  9443-JUBUO    Male              0    True       True      72   
2990  2309-OSFEU    Male              0   False      False      10   
2992  4177-JPDFU    Male              0   False      False      12   
2994  8046-DNVTL    Male              0    True      False      62   
2995  2209-XADXF  Female              0   False      False       1   

     PhoneService MultipleLines InternetService OnlineSecurity  ...  \
0           False          None             DSL          False  ...   
3           False          None             DSL           True  ...   
7           False          None             DSL           True  ...   
20          False          None             DSL          False  ...   
27          False          None             DSL          False  ...   
...           ...           ...             ...            ...  ...   
2972        False          None             DSL           True  ...   
2990        False          None             DSL          False  ...   
2992        False          None             DSL          False  ...   
2994        False          None             DSL           True  ...   
2995        False          None             DSL          False  ...   

     DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
0               False       False       False           False  Month-to-month   
3                True        True       False           False        One year   
7               False       False       False           False  Month-to-month   
20               True       False       False            True  Month-to-month   
27              False       False       False           False  Month-to-month   
...               ...         ...         ...             ...             ...   
2972             True        True        True            True        Two year   
2990            False       False       False            True  Month-to-month   
2992            False        True       False           False  Month-to-month   
2994            False        True       False           False        Two year   
2995            False       False       False           False  Month-to-month   

     PaperlessBilling              PaymentMethod MonthlyCharges  TotalCharges  \
0                True           Electronic check      29.850000     29.850000   
3               False  Bank transfer (automatic)      42.299999   1840.750000   
7               False               Mailed check      29.750000    301.899994   
20               True           Electronic check      39.650002     39.650002   
27              False           Electronic check      30.200001     30.200001   
...               ...                        ...            ...           ...   
2972             True    Credit card (automatic)      65.250000   4478.850098   
2990             True               Mailed check      33.900002    298.450012   
2992            False               Mailed check      34.000000    442.450012   
2994             True    Credit card (automatic)      38.599998   2345.550049   
2995            False  Bank transfer (automatic)      25.250000     25.250000   

      Churn  
0     False  
3     False  
7     False  
20     True  
27     True  
...     ...  
2972  False  
2990   True  
2992  False  
2994  False  
2995  False  

[269 rows x 21 columns]

<span style="color: orange;">**Observation**</span> The 'Multiple Lines' column indicates whether a client has more than one phone line. This information is dependent on whether the client has phone service. Therefore, the null values in the 'Multiple Lines' column will be replaced with the corresponding entry from the 'PhoneService' column, indicating whether the client has phone service or not.

In [42]:
# Checking the unique values in both columns
unique_phone_service = train_df['PhoneService'].unique().tolist()
unique_multi_lines = train_df['MultipleLines'].unique().tolist()
print(f'The unique phone service entries are {unique_phone_service},\nWhile the unique multiline entries are {unique_multi_lines}')

The unique phone service entries are [False, True, 'Yes', 'No'],
While the unique multiline entries are [None, False, True, 'No', 'No phone service', 'Yes']


<span style="color: orange;">**Observation**</span>   
The unique values in the Phone Service column, False, True, 'Yes' and 'No', show a distinct pattern.  
- The first category, True and 'Yes', indicate that the customer has phone service
- The second category, False and 'No', indicate that the client does not have a phone service.  

The unique values in the MultipleLines column, None, False, True, 'No', 'No phone service' and 'Yes', show a distinct pattern.  
- The first category, True and 'Yes', indicate that the customer has Multiple Lines
- The second category, False and 'No', and 'No phone service' indicate that the client does not have a phone service.
- The Third category, 'None', represents the null entries.  

<span style="color: orange;">**Action**</span>
1. Standardize the Phone Service column entries to True and false only by changing the 'No' and 'Yes' entries to False and True respectively. This will have the column with only two conditions: True(Client has Phone Service), False(Client does not have phone service)
2. Use the Phone Service column to standardize the Multiple lines column.
- First convert the 'Yes' to True, then 'No' and 'No phone service' to False.
- Then use the condition in the phone service column false when the entry is 'None' in the Multiple lines column to change that to False. This is under the presumption that, if the Entry is false, in the Phone service(Client does not have phone service), then none means the client cannot have multiple lines. this will standardize the entries in the multiple lines column to only two states, True and False


In [43]:
# Replace 'Yes' with True and 'No' with False
train_df['PhoneService'] = train_df['PhoneService'].replace({'Yes': True, 'No': False})

In [44]:
# Check unique entries in the phone service column
unique_phnsvs = train_df['PhoneService'].unique()
phn_svs_nulls = train_df['PhoneService'].isnull().sum()
print(f'The unique values in the phone Service column are {unique_phnsvs} and there are {phn_svs_nulls} null entries in the column.')

The unique values in the phone Service column are [False  True] and there are 0 null entries in the column.


In [45]:
# Replace 'Yes' with True and 'No' with False and 'No phone service to false'
train_df['MultipleLines'] = train_df['MultipleLines'].replace({'Yes': True, 'No': False, 'No phone service': False, 'None': False})
# Check unique entries in the phone service column
train_df['MultipleLines'].unique()

array([None, False, True], dtype=object)

In [46]:
# Fill null entries in 'MultipleLines' based on 'PhoneService' condition
train_df.loc[train_df['PhoneService'] == False, 'MultipleLines'] = train_df['MultipleLines'].fillna(False)
# Check new null entries in Multiple lines column
print(f'The are now {train_df['MultipleLines'].isnull().sum()} null entries in the MultipleLines column and the unique values are {train_df['MultipleLines'].unique()}')

The are now 0 null entries in the MultipleLines column and the unique values are [False True]


In [47]:
# Checking new train_df null status
train_df.isna().sum().sort_values(ascending=False)

OnlineBackup        651
OnlineSecurity      651
StreamingMovies     651
StreamingTV         651
TechSupport         651
DeviceProtection    651
Churn                 1
MultipleLines         0
InternetService       0
PhoneService          0
gender                0
tenure                0
Dependents            0
Partner               0
SeniorCitizen         0
Contract              0
PaperlessBilling      0
PaymentMethod         0
MonthlyCharges        0
TotalCharges          0
customerID            0
dtype: int64

#### **Cleaning the OnlineBackup OnlineSecurity ,StreamingMovies, StreamingTV, TechSupport, DeviceProtection Columns**
<span style="color: orange;">**Observation**</span>  As observed in the null counts for the train_df data-frame, all 6 columns listed above have 651 null entries each. With this, the data-frame understanding shows that these column entries are as a result of the client having internet service or not and thus the entries in this columns are as a result of the entries in the InternetService column. All the null entries are directly related to the 'No' entry in the 'InternetService' column. Thus we will fill the nulls with false since the client does not have Internes service('No'), and thus these other services cannot be accessed from the company.

In [48]:
train_df['InternetService'].unique()

array(['DSL', 'Fiber optic', 'No'], dtype=object)

In [52]:
# Filling the nulls in the six columns with the condition the 'No' from 'InternetService'
train_df.loc[train_df['InternetService'] == 'No', ['OnlineBackup', 'OnlineSecurity', 'StreamingMovies', 'StreamingTV', 'TechSupport', 'DeviceProtection']] = train_df.loc[train_df['InternetService'] == 'No', ['OnlineBackup', 'OnlineSecurity', 'StreamingMovies', 'StreamingTV', 'TechSupport', 'DeviceProtection']].fillna(False)

In [53]:
unique_values = train_df[['OnlineBackup', 'OnlineSecurity', 'StreamingMovies', 'StreamingTV', 'TechSupport', 'DeviceProtection']].apply(pd.Series.unique)

unique_values

OnlineBackup       OnlineSecurity      StreamingMovies  \
0                 True                False                False   
1                False                 True                 True   
2                   No                   No                   No   
3                  Yes                  Yes                  Yes   
4  No internet service  No internet service  No internet service   

           StreamingTV          TechSupport     DeviceProtection  
0                False                False                False  
1                 True                 True                 True  
2                   No                  Yes                   No  
3                  Yes                   No                  Yes  
4  No internet service  No internet service  No internet service

In [55]:
# Define a function to clean unique values in columns
def clean_unique_values(column):
    mapping = {'Yes': True, 'No': False, 'No internet service': False}
    return column.replace(mapping).unique()

# Clean unique values for each specified column using the function
clean_backup = clean_unique_values(train_df['OnlineBackup'])
clean_security = clean_unique_values(train_df['OnlineSecurity'])
clean_movies = clean_unique_values(train_df['StreamingMovies'])
clean_tv = clean_unique_values(train_df['StreamingTV'])
clean_tech_support = clean_unique_values(train_df['TechSupport'])
clean_protection = clean_unique_values(train_df['DeviceProtection'])

print("Cleaned unique values for OnlineBackup:", clean_backup)
print("Cleaned unique values for OnlineSecurity:", clean_security)
print("Cleaned unique values for StreamingMovies:", clean_movies)
print("Cleaned unique values for StreamingTV:", clean_tv)
print("Cleaned unique values for TechSupport:", clean_tech_support)
print("Cleaned unique values for DeviceProtection:", clean_protection)

Cleaned unique values for OnlineBackup: [ True False]
Cleaned unique values for OnlineSecurity: [False  True]
Cleaned unique values for StreamingMovies: [False  True]
Cleaned unique values for StreamingTV: [False  True]
Cleaned unique values for TechSupport: [False  True]
Cleaned unique values for DeviceProtection: [False  True]


In [56]:
#Checking new train_df null status
train_df.isna().sum().sort_values(ascending=False)

Churn               1
DeviceProtection    0
TotalCharges        0
MonthlyCharges      0
PaymentMethod       0
PaperlessBilling    0
Contract            0
StreamingMovies     0
StreamingTV         0
TechSupport         0
customerID          0
gender              0
OnlineSecurity      0
InternetService     0
MultipleLines       0
PhoneService        0
tenure              0
Dependents          0
Partner             0
SeniorCitizen       0
OnlineBackup        0
dtype: int64

<span style="color:red;">This text will appear red</span>

<span style="color: orange;">**This text will appear orange**</span>